In [2]:
import os
import pandas as pd


In [3]:
os.chdir('C:/Users/hp/Desktop/tutorial/kaim-week-5')
print(os.getcwd())

C:\Users\hp\Desktop\tutorial\kaim-week-5


In [4]:
%load_ext autoreload
%autoreload 2

In [5]:
from telethon import TelegramClient
import csv
import os

In [7]:
from dotenv import load_dotenv
from scripts.scraping import scraper,main
load_dotenv('.env')
api_id = os.getenv('App_API_ID')
api_hash = os.getenv('App_API_HASH')
phone = os.getenv('phone')
# Remove the session file if it exists
if os.path.exists('scraping_session.session'):
    os.remove('scraping_session.session')
# Initialize the client once
client = TelegramClient('scraping_session', api_id, api_hash)
async def run():
    async with client:
        await main(client)  # Call the scraping main function#

# Run the asynchronous function by awaiting directly in the notebook
await run()

#async def main():
    # Start the client and log in
#    await client.start(phone)

    # After logging in, you can start using the client for scraping data or other actions

#    print("Logged in successfully!")

# Run the async function
#await main()

Signed in successfully as Basleal Abera; remember to not break the ToS or you will risk an account ban!
Scraped data from @ZemenExpress
Scraped data from @nevacomputer
Scraped data from @meneshayeofficial
Scraped data from @ethio_brand_collection
Scraped data from @Leyueqa


In [8]:
from scripts.preprocessing import load_data
path='C:/Users/hp/Desktop/tutorial/kaim-week-5/data/telegram_data.csv'
data=load_data(path)
data.head()

Channel Title Channel Username    ID  \
0  Zemen Express®    @ZemenExpress  6035   
1  Zemen Express®    @ZemenExpress  6034   
2  Zemen Express®    @ZemenExpress  6033   
3  Zemen Express®    @ZemenExpress  6032   
4  Zemen Express®    @ZemenExpress  6031   

                                             Message  \
0  💥💥...................................💥💥\n\n📌Do...   
1  💥💥...................................💥💥\n\n📌Do...   
2  💥💥...................................💥💥\n\n📌Do...   
3  💥💥...................................💥💥\n\n📌 H...   
4  💥💥...................................💥💥\n\n📌 H...   

                        Date  Media Path  
0  2025-01-18 05:12:07+00:00         NaN  
1  2025-01-18 05:11:55+00:00         NaN  
2  2025-01-18 05:11:36+00:00         NaN  
3  2025-01-17 18:02:23+00:00         NaN  
4  2025-01-17 18:02:11+00:00         NaN

In [9]:
data.isnull().sum()

Channel Title          0
Channel Username       0
ID                     0
Message             1023
Date                   0
Media Path          3000
dtype: int64

Our message column in our data has 1023 missing values

In [18]:
from scripts.preprocessing import preprocessing_data, remove_emoji
data = preprocessing_data(data)
data.isnull().sum()

Channel Title          0
Channel Username       0
ID                     0
Message                0
Date                   0
Media Path          1977
Labeled_Message        0
dtype: int64

We have successfully cleaned our data

In [20]:
# Apply the function to the 'Message' column
data['Message'] = data['Message'].apply(remove_emoji)
import re
# Remove the sequence of dots (or similar patterns) from the 'Message' and 'labeled_message' column
data['Message'] = data['Message'].apply(lambda x: re.sub(r'\.{2,}', '', x))
data['Labeled_Message'] = data['Labeled_Message'].apply(lambda x: re.sub(r'\.{2,}', '', x))
data.head()

Channel Title Channel Username    ID  \
0  Zemen Express®    @ZemenExpress  6035   
1  Zemen Express®    @ZemenExpress  6034   
2  Zemen Express®    @ZemenExpress  6033   
3  Zemen Express®    @ZemenExpress  6032   
4  Zemen Express®    @ZemenExpress  6031   

                                             Message  \
0  \n\nDoor Bottom Seal Strip Stopper\nበበርዎ ስር አቧ...   
1  \n\nDoor Bottom Seal Strip Stopper\nበበርዎ ስር አቧ...   
2  \n\nDoor Bottom Seal Strip Stopper\nበበርዎ ስር አቧ...   
3  \n\n High pressure water Gun nozzle\n\n   መኪና ...   
4  \n\n High pressure water Gun nozzle\n\n   መኪና ...   

                        Date  Media Path  \
0  2025-01-18 05:12:07+00:00         NaN   
1  2025-01-18 05:11:55+00:00         NaN   
2  2025-01-18 05:11:36+00:00         NaN   
3  2025-01-17 18:02:23+00:00         NaN   
4  2025-01-17 18:02:11+00:00         NaN   

                                     Labeled_Message  
0   B-PRODUCT\nDoor O\nBottom O\nSeal O\nStrip O\...  
1   B-PRODUCT\nDoor O\nBottom O\nSeal O\nStrip O\...  
2   B-PRODUCT\nDoor O\nBottom O\nSeal O\nStrip O\...  
3   B-PRODUCT\nHigh O\npressure O\nwater O\nGun O...  
4   B-PRODUCT\nHigh O\npressure O\nwater O\nGun O...

In [21]:
#save the cleaned telegram data 
data.to_csv('clean_telegram_data.csv')

In [24]:
from scripts.preprocessing import label_message_utf8_with_birr
# Apply the updated function to the non-null messages
data['Labeled_Message'] = data['Message'].apply(label_message_utf8_with_birr)
data.head(10)['Labeled_Message']

0    Door O\nBottom O\nSeal O\nStrip O\nStopper O\n...
1    Door O\nBottom O\nSeal O\nStrip O\nStopper O\n...
2    Door O\nBottom O\nSeal O\nStrip O\nStopper O\n...
3    High O\npressure O\nwater O\nGun O\nnozzle O\n...
4    High O\npressure O\nwater O\nGun O\nnozzle O\n...
5    High O\npressure O\nwater O\nGun O\nnozzle O\n...
6    Electric O\nHair O\nSteamer O\nCap O\nHigh O\n...
7    Electric O\nHair O\nSteamer O\nCap O\nHigh O\n...
8    Straight O\nComb O\nHair O\nstylist O\n‼ምንም O\...
9    Straight O\nComb O\nHair O\nstylist O\n‼ምንም O\...
Name: Labeled_Message, dtype: object

In [25]:
# Save the updated labeled dataset to a file in CoNLL format
labeled_data_birr_path = 'labeled_telegram_product_price_location.txt-' # Define the path where the labeled data will be saved
# Open the file for writing with UTF-8 encoding
with open(labeled_data_birr_path, 'w', encoding='utf-8') as f:
    for index, row in data.iterrows():
        # Write the labeled message from the row to the file
        f.write(f"{row['Labeled_Message']}\n\n")